File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int l = 0, r = 1;
        int res = 0;
        while(r<prices.size()){
            if(prices[l] < prices[r]){
            int prof = prices[r]-prices[l];
            res = max(res, prof);
            }
            else{
                l = r;
            }
            r++;
        }
        return res;
    }
};

File: Data Structures & Algorithms/capacity-to-ship-packages-within-d-days/submission-1.py
class Solution:
    def shipWithinDays(self, weights: List[int], days: int) -> int:
        l, r = max(weights), sum(weights)
        res = 0
        while l <= r:
            mid = (l+r) // 2
            i = 0
            count = 1
            tmp = mid
            while i < len(weights):
                if mid - weights[i] < 0:
                    count += 1
                    mid = tmp
                mid -= weights[i]
                i+=1
            mid = tmp
            if count <= days:
                res = mid
                r = mid - 1
            else:
                l = mid +1
        return res